### PR TITLE
Test: Refine annotaion and test code for sensor subsys.

### PR DESCRIPTION
**Pull request by youhuazhu**
1. Add more detail info to make the purpose and process
of the test cases more clear which include test goal,
test step, input, judging criteria, constraints, etc.,
and these can be seen in our Zephyr documentations.
2. Add some negative test code.

Signed-off-by: YouhuaX Zhu youhuax.zhu@intel.com

### CODE REVIEW

File: tests/drivers/sensor/generic/src/main.c [hunk 1]
```diff
@@ -108,25 +141,54 @@ static void trigger_handler(const struct device *dev,
...
+ * then check the result.
+ * -# Handle different types of triggers, based on time, data,threshold,
+ * based on a delta value, near/far events and single/double tap and
+ * check the result.
+ *
+ * Expected Test Result:
+ * - Application can get multiple channels for dummy sensor.
+ *
+ * Pass/Fail Criteria:
+ * - Successful if the check of step 1/2/3 are all pass.
```
Comment (LeiW000) [inline tests/drivers/sensor/generic/src/main.c:176]: "Pass" should be "Passed".

File: tests/drivers/sensor/generic/src/main.c [hunk 2]
```diff
@@ -62,18 +62,47 @@ static struct trigger_sequence trigger_elements[] = {
...
+ * -# Get SENSOR_CHAN_LIGHT/SENSOR_CHAN_RED/SENSOR_CHAN_GREEN/
+ * SENSOR_CHAN_BLUE/SENSOR_CHAN_BLUE channels from the sensor,
+ * and check the result.
+ *
+ * Expected Test Result:
+ * - Application can get multiple channels for dummy sensor.
  *
- * @see sensor_sample_fetch(). sensor_channel_get()
+ * Pass/Fail Criteria:
+ * - Successful if the check of step 2/3 are all pass.
```
Comment (LeiW000) [inline tests/drivers/sensor/generic/src/main.c:99]: Same as line 176.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/drivers/sensor/generic/src/main.c (hunk 1 above)
Comment (youhuazhu) [reply tests/drivers/sensor/generic/src/main.c]: fixed,  thanks.

File: tests/drivers/sensor/generic/src/main.c (hunk 2 above)
Comment (youhuazhu) [reply tests/drivers/sensor/generic/src/main.c]: fixed,  thanks.

**Review by LeiW000 — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Comment by LeiW000** [comment]
Is there anything other comment? If no, can anyone help to merge it?

**Comment by LeiW000** [comment]
Does anyone help to merge it? There already are two PRs.

File: tests/drivers/sensor/generic/src/dummy_sensor.c [hunk 3]
```diff
@@ -78,6 +79,10 @@ static int dummy_sensor_init(const struct device *dev)
 	/* i2c should be null for dummy driver */
 	const struct device *i2c = device_get_binding(config->i2c_name);
 
+	/* Bus and address should be configurated */
```
Comment (nashif) [inline tests/drivers/sensor/generic/src/dummy_sensor.c:82]: configured

Comment (youhuazhu) [reply tests/drivers/sensor/generic/src/dummy_sensor.c]: fixed, Thanks for reminding.

**Pull request merged**